lwgenerate/aws.go

```diff
@@ -0,0 +1,294 @@
+package lwgenerate
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/hcl/v2/hclwrite"
+	"github.com/pkg/errors"
+)
+
+type ExistingIamRoleDetails struct {
+	// Existing IAM Role ARN
+	Arn string
+
+	// Existing IAM Role Name
+	Name string
+
+	// Existing IAM Role External Id
+	ExternalId string
```

Comment:
Are these fields required to be exported?

Nit: Also perhaps not all fields require comments. I don't believe we follow this pattern elsewhere.

Reply:
Similar to above, these get accessed outside lwgenerate package.

Reply:
Re doc strings; I can remove these

Reply:
I disagree with @dmurray-lacework nitpick, comments are great and we should keep and encourage them.

Example that we do this here: https://github.com/lacework/go-sdk/blob/main/cli/cmd/compliance.go#L35-L60